src/helm/benchmark/metrics/bhasa_metrics.py

```diff
@@ -0,0 +1,314 @@
+        inp: str = self._remove_braces(request_state.instance.input.text)
+
+        assert request_state.result is not None
+        pred: str = self._remove_braces(request_state.result.completions[0].text.strip())
+
+        result: List[Stat] = []
+
+        # Compute ChrF++ metrics
+        result.extend([Stat(MetricName(name)).add(float(val)) for name, val in self._compute_chrf(refs, pred).items()])
+
+        return result
+
+class BhasaQAMetric(Metric):
+    """Bhasa QA Metrics
+
+    This class computes the following standard SQuAD v1.1 metrics
+
+    1. SQuAD exact match
+    2. SQuAD macro-averaged F1 score
+
+    @inproceedings{rajpurkar-etal-2016-squad,
+        title = "{SQ}u{AD}: 100,000+ Questions for Machine Comprehension of Text",
+        author = "Rajpurkar, Pranav  and
+            Zhang, Jian  and
+            Lopyrev, Konstantin  and
+            Liang, Percy",
+        editor = "Su, Jian  and
+            Duh, Kevin  and
+            Carreras, Xavier",
+        booktitle = "Proceedings of the 2016 Conference on Empirical Methods in Natural Language Processing",
+        month = nov,
+        year = "2016",
+        address = "Austin, Texas",
+        publisher = "Association for Computational Linguistics",
+        url = "https://aclanthology.org/D16-1264",
+        doi = "10.18653/v1/D16-1264",
+        pages = "2383--2392",
+    }
+    """
+
+    def __init__(self, language: str = 'en'):
+        self.language: str = language
+        self.metrics: Dict[str, Callable] = {
+            "squad_exact_match": exact_match,
+            "squad_f1_score": self.squad_f1_score,
+        }
+
+    def evaluate(
+        self, scenario_state: ScenarioState, metric_service: MetricService, eval_cache_path: str, parallelism: int
+    ) -> MetricResult:
+        return super().evaluate(scenario_state, metric_service, eval_cache_path, parallelism=parallelism)
+
+    def split_text(self, text: str) -> List[str]:
+        """
+        For Thai, this will split the text using PyThaiNLP's tokenizer.
+        For all other languages, this will:
+        - Lower text
+        - Remove punctuation
+        - Remove extra whitespace
+
+        If the language is English, it will
+        - Remove articles "a", "an", and "the"
+
+        Modifies code from [QuAC](http://quac.ai/) found at https://s3.amazonaws.com/my89public/quac/scorer.py
+        """
+
+        def remove_articles(text: str) -> str:
+            return re.sub(r"\b(a|an|the)\b", " ", text)
+
+        def white_space_fix(text: str) -> str:
+            return " ".join(text.split())
+
+        def remove_punc(text: str) -> str:
+            exclude = set(string.punctuation)
+            return "".join(ch for ch in text if ch not in exclude)
+
+        def lower(text: str) -> str:
+            return text.lower()
+
+        normalized_text = remove_punc(lower(text))
+        if self.language == "th":
+            return set(word_tokenize(normalized_text, engine="newmm"))
```

Comment:
@raileymontalan This should not be a set right? We do want to count the same token multiple times if it does indeed occur multiple times.

Reply:
@weiqipedia Resolved in my latest commit.